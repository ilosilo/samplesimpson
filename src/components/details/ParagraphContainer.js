import * as React from 'react';
import { useState } from 'react';
import { textColor, addButton, listItem } from '../../config/color';

import {
    ScrollView,
    StyleSheet,
    Text,
    TextInput,
    TouchableOpacity,
    Modal,
    View,
} from 'react-native';
import { saveItem, deleteItem } from '../../config/Storage';

const ParagraphContainer = (props) => {

    const [paragraph, setParagraph] = useState(props.item.about);
    const [changeParagraph, setChangeParagraph] = useState(false)
    const { item } = props;
    const newItem = {
        id: '',
        name: '',
        avatar: '',
        job: '',
        about: ''
    }

    const setNewParagraph = () => {
        newItem.id = item.id;
        newItem.name = item.name;
        newItem.avatar = item.avatar;
        newItem.job = item.job;
        newItem.about = paragraph;
        deleteItem(item).then(() => (saveItem(newItem)));
        item.about = paragraph;
        setChangeParagraph(false);
    }

    return (
        <ScrollView style={st.paragraphContainerStyle}>
            <TouchableOpacity onPress={() => setChangeParagraph(true)}>
                {
                    item.about ?
                        <Text style={st.paragraphTextStyle}>{item.about}</Text>
                        :
                        <Text style={st.paragraphTextStyle}>There is no more information</Text>
                }
            </TouchableOpacity>
            <Modal
                animationType="slide"
                transparent={true}
                visible={changeParagraph}
                onRequestClose={() => {
                    setChangeParagraph(!changeParagraph);
                }}
            >
                <View style={st.modalStyle}>
                    <Text> About: </Text>
                    <TextInput style={st.input}
                        multiline
                        numberOfLines={10}
                        value={paragraph}
                        onChangeText={text => setParagraph(text)}></TextInput>
                    <View style={{ flexDirection: 'row' }}>
                        <TouchableOpacity
                            onPress={() => { setNewParagraph() }}
                            style={st.boxStyle}>
                            <Text> Change About </Text>
                        </TouchableOpacity>
                        <TouchableOpacity
                            onPress={() => setChangeParagraph(false)}
                            style={st.boxStyle}>
                            <Text> Cancel </Text>
                        </TouchableOpacity>
                    </View>

                </View>
            </Modal>
        </ScrollView>
    );
};

export default ParagraphContainer;

const st = StyleSheet.create({
    input: {
        width: '100%',
        borderBottomColor: addButton,
        borderBottomWidth: 2,
        borderColor: listItem,
        borderWidth: 1,
        color: 'black'
    },
    modalStyle: {
        backgroundColor: 'white',
        width: '90%',
        height: 350,
        alignSelf: 'center',
        justifyContent: 'center',
        alignItems: 'center',
        position: 'absolute',
        top: 50,
        borderRadius: 30,
        borderRightWidth: 2,
        borderBottomWidth: 2,
        borderColor: 'gray',
        padding: 30
    },
    boxStyle: {
        borderTopWidth: 2,
        borderLeftWidth: 2,
        borderRightWidth: 2,
        borderColor: 'gray',
        alignItems: 'center',
        marginTop: 30,
        borderRadius: 30,
        padding: 10,
        width: '50%'
    },
    input: {
        borderWidth: 1,
        borderColor: 'lightgray',
        width: '90%'
    },
    paragraphContainerStyle: {
        alignSelf: 'center',
        alignContent: 'center',
        padding: 10
    },
    paragraphTextStyle: {
        fontSize: 15,
        flexWrap: 'wrap',
        alignSelf: 'center',
        color: textColor
    },
});
