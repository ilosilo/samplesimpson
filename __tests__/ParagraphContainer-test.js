
 import 'react-native';
 import React from 'react';
 import ParagraphContainer from '../src/components/details/ParagraphContainer';
 import renderer from 'react-test-renderer';
 
 test('renders correctly', () => {
    const item = {"id":"3","name":"Bart Simpson","avatar":"https://static.wikia.nocookie.net/simpsons/images/6/65/Bart_Simpson.png/revision/latest/scale-to-width-down/250?cb=20190409004756","job":"Student","about":"Bartholomew JoJo \"Bart\" Simpson "}
    
    const tree = renderer.create(<ParagraphContainer item={item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });