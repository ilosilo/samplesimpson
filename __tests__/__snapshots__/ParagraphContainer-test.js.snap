// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "alignContent": "center",
      "alignSelf": "center",
      "padding": 10,
    }
  }
>
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "rgb(84, 75, 63)",
            "flexWrap": "wrap",
            "fontSize": 15,
          }
        }
      >
        Bartholomew JoJo "Bart" Simpson 
      </Text>
    </View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderBottomWidth": 2,
            "borderColor": "gray",
            "borderRadius": 30,
            "borderRightWidth": 2,
            "height": 350,
            "justifyContent": "center",
            "padding": 30,
            "position": "absolute",
            "top": 50,
            "width": "90%",
          }
        }
      >
        <Text>
           About: 
        </Text>
        <TextInput
          multiline={true}
          numberOfLines={10}
          onChangeText={[Function]}
          style={
            Object {
              "borderColor": "lightgray",
              "borderWidth": 1,
              "width": "90%",
            }
          }
          value="Bartholomew JoJo \\"Bart\\" Simpson "
        />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "gray",
                "borderLeftWidth": 2,
                "borderRadius": 30,
                "borderRightWidth": 2,
                "borderTopWidth": 2,
                "marginTop": 30,
                "opacity": 1,
                "padding": 10,
                "width": "50%",
              }
            }
          >
            <Text>
               Change About 
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "gray",
                "borderLeftWidth": 2,
                "borderRadius": 30,
                "borderRightWidth": 2,
                "borderTopWidth": 2,
                "marginTop": 30,
                "opacity": 1,
                "padding": 10,
                "width": "50%",
              }
            }
          >
            <Text>
               Cancel 
            </Text>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</RCTScrollView>
`;
